torch/distributed/fsdp/_init_utils.py

```diff
@@ -753,6 +753,12 @@ def _check_single_device_module(
     module must be either fully on the CPU or fully on a non-CPU device.
     """
     devices = {param.device for param in _get_orig_params(module, ignored_params)}
+    if len(devices) == 2 and torch.device("cpu") in devices:
+        warnings.warn(
+            f"FSDP wrapping module with params on multiple devices: {devices}"
+            "This could occur in cases where another parallelism algorithm has already"
+            "moved some parameters to a non-CPU device."
+        )
     if len(devices) > 1:
```

Comment:
If `if len(devices) == 2 and torch.device("cpu") in devices == True`, I am not seeing why this `if len(devices) > 1:` branch would not also get taken, still leading to the `RuntimeError` getting raised.

Reply:
yeah, we need to change this to an `else`